### PR TITLE
Update guidance for F# lambda's.

### LINKED ISSUE #25207: Guidance for MatchLambda as argument
Hello, in the section [Formatting function parameter application](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-function-parameter-application) it is mentioned that

> Parameters should generally be indented relative to the function or fun/function keyword, regardless of the context in which the

However, there is no example given for `function | Some _ -> x | None -> y` expressions.
Consider the following code:
```fsharp
module Foo =
    let bar =
        []
        |> List.choose 
            (function
             | _ -> "")
```

Would it be formatted like this or follow the rule of [Formatting constructors, static members, and member invocations](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-constructors-static-members-and-member-invocations)?

Something like
```fsharp
module Foo =
    let bar =
        []
        |> List.choose (
            function
            | _ -> ""
        )
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8c09996f-24ac-cf39-ef43-d7909cbc2b89
* Version Independent ID: 1efdd523-8552-6421-e149-74f76826ef8f
* Content: [F# code formatting guidelines](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting)
* Content Source: [docs/fsharp/style-guide/formatting.md](https://github.com/dotnet/docs/blob/main/docs/fsharp/style-guide/formatting.md)
* Product: **dotnet-fsharp**
* GitHub Login: @cartermp
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by nojaf**
Fixes #25207
One TODO left, what to do with https://github.com/dotnet/docs/issues/22834 @dsyme?

### CODE REVIEW

File: docs/fsharp/style-guide/formatting.md [hunk 1]
```diff
@@ -1069,24 +1067,61 @@ let printVolumes x =
...
+    arg2 
+    arg3 
+    (function
+     | Choice1of2 x
+     | Choice2of2 y)
 ```
 
 If the body of a lambda expression is multiple lines long, you should consider refactoring it into a locally-scoped function.
 
+// TODO: Is https://github.com/dotnet/docs/issues/22834 still relevant?
```
Comment (dsyme) [inline docs/fsharp/style-guide/formatting.md:1123]: I'm not sure of the question here?

**Comment by dsyme** [comment]
> One TODO left, what to do with #22834 @dsyme?

Yes, not sure what you're asking here. It's about some two-space indentation samples?  Could you list them below so I know exactly what's being discussed?  thanks

File: docs/fsharp/style-guide/formatting.md (hunk 1 above)
Comment (nojaf) [reply docs/fsharp/style-guide/formatting.md:1123]: Sorry @dsyme, the concrete question would be if this indent from the function name rule still applies when the lambda is the last argument.
Imagine if the indent size is 2 spaces.
```fsharp
list1
|> List.iter (fun elem ->
  // should the body start one indent from the pipe
      // or one indent from the `List.iter`?
  printfn $"A very long line to format the value: %d{elem}")
```

Comment (nojaf) [reply docs/fsharp/style-guide/formatting.md]: Thanks, ready for review now.

**Review by dsyme — APPROVED** [review]

**Comment by dsyme** [comment]
Thank you!!

**Pull request merged**